docs/how-tos/upload-module.md

```diff
@@ -0,0 +1,276 @@
+---
+title: "How to upload a module"
+linkTitle: "Upload a module"
+type: "docs"
+weight: 26
+images: ["/registry/module-puzzle-piece.svg"]
+tags: ["modular resources", "components", "services", "registry"]
+description: "Make your module available to others by uploading it to the modular resource registry."
+aliases:
+  - /registry/upload/
+  - /extend/modular-resources/upload/
+  - /modular-resources/upload/
+---
+
+After [writing a module](/how-tos/create-module/), you can upload your module to the Viam registry to:
+
+- share your module with other Viam users
+- deploy your module to a fleet of machines from a central interface
+
+You can choose to upload it to the Viam registry as a _public module_ that is shared with other Viam users, or as a _private module_ that is shared only within your [organization](/cloud/organizations/).
+
+To upload your custom module to the [Viam registry](https://app.viam.com/registry), either as a public or private module, use the Viam CLI commands `create` and `upload` following these instructions:
+
+{{< table >}}
+{{% tablestep %}}
+**1. Install the CLI**
+
+First, [install the Viam CLI](/cli/#install) and [authenticate](/cli/#authenticate) to Viam, from the same machine that you intend to upload your module from.
+
```

Comment:
we should propably make the instructions an include so that they can be included rather than linking out

Reply:
Actually this should probably also be a prerequisite. 

And you can use this:

```
{{< readfile "/static/include/how-to/install-cli.md" >}}
```

There's currently an extra sentence in it but I am removing that in DOCS-2794

Reply:
merging that here https://github.com/viamrobotics/docs/pull/3291 for you